src/App.jsx

```diff
@@ -14,33 +16,132 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
+  const [isReversed, setIsReversed] = useState(false);
+  const [activeButton, setActiveButton] = useState(null);
+  const [showReset, setShowReset] = useState(false);
+  const [sortedByLength, setSortedByLength] = useState(false);
+  const [reverseCount, setReverseCount] = useState(0);
```

Comment:
```suggestion
  const [isReversed, setIsReversed] = useState(false);
  const [sortedBy, setSortedBy] = useState('');  // '' or 'length' or 'abc'
```
why do you need so many states? enough only two.
based on this states manage classes for buttons, visibility of reset button, sort and reverse or not goodsFromServer and render result.